### PR TITLE
change: update repo with gui, free game and popup refactor

### PR DESCRIPTION
**Pull request by GabrielNied**

### CODE REVIEW

File: assets/scripts/broker/titan/GameTitanResponses.ts [hunk 1]
```diff
@@ -15,20 +16,23 @@ export default class GameTitanResponses extends cc.Component {
   static async getConfig(): Promise<Config> {
     const titan = getTitan();
     const request = makeRefreshPacket(titan.gameApi);
+    console.log('# request: ', request);
```
Comment (stefanhacks) [inline assets/scripts/broker/titan/GameTitanResponses.ts:19]:
```suggestion
```

File: assets/scripts/broker/titan/GameTitanResponses.ts [hunk 2]
```diff
@@ -15,20 +16,23 @@ export default class GameTitanResponses extends cc.Component {
...
     const response = await titan.gameApi.gamecomms.sendPacketJson<XMLPlay>(request);
     return (await formatXMLConfig(response)) as Config;
   }
 
-  static async getPlay(freeGame?: FreeGameOffer): Promise<BaseGameRound> {
+  static async getPlay(freeGameOffer?: FreeGameOffer): Promise<BaseGameRound> {
     const titan = getTitan();
-    const request = makeSpinPacket(titan.gameApi, freeGame);
+    const request = makeSpinPacket(titan.gameApi, freeGameOffer);
+    console.log('# request: ', request);
```
Comment (stefanhacks) [inline assets/scripts/broker/titan/GameTitanResponses.ts:27]:
```suggestion
```

File: assets/scripts/broker/titan/GameTitanResponses.ts [hunk 3]
```diff
@@ -15,20 +16,23 @@ export default class GameTitanResponses extends cc.Component {
...
     const response = await titan.gameApi.gamecomms.sendPacketJson<XMLPlay>(request);
     return (await formatXMLPlay(response)) as BaseGameRound;
   }
 
   static async getFreeSpin(): Promise<BaseGameRound> {
     const titan = getTitan();
     const request = makeFreeSpinPacket(titan.gameApi);
+    console.log('# request: ', request);
```
Comment (stefanhacks) [inline assets/scripts/broker/titan/GameTitanResponses.ts:35]:
```suggestion
```

File: assets/scripts/broker/titan/GameTitanResponses.ts [hunk 4]
```diff
@@ -42,13 +46,15 @@ export default class GameTitanResponses extends cc.Component {
       numChips,
       activePaylines: '0',
     });
+    console.log('# request: ', request);
```
Comment (stefanhacks) [inline assets/scripts/broker/titan/GameTitanResponses.ts:49]:
```suggestion
```

File: assets/scripts/broker/titan/GameTitanResponses.ts [hunk 5]
```diff
@@ -42,13 +46,15 @@ export default class GameTitanResponses extends cc.Component {
...
     const response = await titan.gameApi.gamecomms.sendPacketJson<XMLPlay>(request);
     return (await formatXMLPlay(response)) as BaseGameRound;
   }
 
-  static async getRejectFreeGames(freeGames: Array<FreeGameOffer>): Promise<RejectFreeGames> {
+  static async getRejectFreeGames(freeGameOffer: FreeGameOffer): Promise<RejectFreeGames> {
     const titan = getTitan();
-    const request = makeRejectFreeGamesPacket(titan.gameApi, freeGames);
+    const request = makeRejectFreeGamesPacket(titan.gameApi, freeGameOffer);
+    console.log('# request: ', request);
```
Comment (stefanhacks) [inline assets/scripts/broker/titan/GameTitanResponses.ts:57]:
```suggestion
```

File: assets/scripts/broker/xml_translator/NetworkTranslator.ts [hunk 6]
```diff
@@ -77,7 +78,7 @@ export function formatXMLPlay(XMLResponse: XMLPlay, fromConfig = false): BaseGam
   console.log('formatXMLPlay responseJson: ', XMLResponse);
```
Comment (stefanhacks) [inline assets/scripts/broker/xml_translator/NetworkTranslator.ts:78]:
```suggestion
```

File: assets/scripts/broker/xml_translator/NetworkTranslator.ts [hunk 7]
```diff
@@ -88,11 +89,10 @@ export function formatXMLPlay(XMLResponse: XMLPlay, fromConfig = false): BaseGam
   translatedPlay.response.tokens = formatTokens(Tokens?.Token);
   translatedPlay.response.inGameProgress = formatGameProgressTokens(InGameProgress?.IGP);
 
-  if (FreeGames) translatedPlay.response.freeGames = formatFreeGames(FreeGames, fromConfig);
-
   // OURS
   translatedPlay.round = formatRound(translatedPlay);
   translatedPlay.stagesInfo = formatStages(Slot);
+  translatedPlay.freeGame = formatFreeGame(translatedPlay.response.freeGames, fromConfig);
   console.log('translatedPlay: ', translatedPlay);
```
Comment (stefanhacks) [inline assets/scripts/broker/xml_translator/NetworkTranslator.ts:96]:
```suggestion
```

File: assets/scripts/state_machine/GameContext.ts [hunk 8]
```diff
@@ -125,25 +125,43 @@ export default class GameContext extends StateContext implements Bootable {
...
 
-  private hasFreeGame(freeGames: FreeGame): boolean {
-    return freeGames && freeGames.offers.length > 0;
+  private handlePracticeMode(): void {
+    this.gameHandler.changeGUIState(ButtonState.Enable);
+    this.assets.UI.playRealButton.setup();
+  }
+
+  private hasFreeGame(config: Config): boolean {
+    console.log('$  GameContext  hasFreeGame  config', config);
```
Comment (stefanhacks) [inline assets/scripts/state_machine/GameContext.ts:134]:
```suggestion
```

**Review by stefanhacks — APPROVED** [review]
Unsure if we're leaving comments in pushes. Disregard suggestions if we are. :)

File: assets/scripts/broker/titan/GameTitanResponses.ts (hunk 1 above)
Comment (GabrielNied) [reply assets/scripts/broker/titan/GameTitanResponses.ts:19]: It's purposeful

**Pull request merged**